CosmicReco/src/PDFFit.cc

```diff
@@ -336,12 +336,66 @@ double GaussianDriftFit::DOCAresidual(ComboHit const& sh, const std::vector<doub
   Straw const& straw = tracker->getStraw(sh.strawId());
   TwoLinePCA pca(straw.getMidPoint(), straw.getDirection(), intercept, dir);
   double traj_time = ((pca.point2() - intercept).dot(dir))/299.9;
-  double hit_t0 = sh.propTime() + traj_time + t0 + srep.driftTimeOffset(sh.strawId(), 0, 0, pca.dca());
 
   double predictedDistance = pca.dca();
+  double hit_t0 = sh.propTime() + traj_time + t0 + srep.driftTimeOffset(sh.strawId(), 0, 0, pca.dca());
   double measuredDistance = srep.driftTimeToDistance(sh.strawId(), sh.time()-hit_t0, 0);
 
-  return predictedDistance-measuredDistance;
+  double resid = predictedDistance - measuredDistance;
+
+  return (pca.s1() > 0 ? resid : -resid);
```

Comment:
I'm not sure I follow this line, is it supposed to be signing by the ambiguity?

Reply:
Yes. I meant to ask you about this further but didn't get around to it:

This is based on `TwoLinePCA_XYZ::LRambig()`, which uses `s2() > 0 ? 1 : -1` but I use `s1()` instead of `s2()` here because TwoLinePCA in the time fit is constructed like this: `TwoLinePCA(straw_p, straw_d, track_p, track_d)` while earlier use of TwoLinePCA_XYZ in CosmicReco is constructed like this `TwoLinePCA_XYZ(track_p, track_d, straw_p, straw_d)`.

I sign the residual because if I only sign the predictedDistance I can get large time residuals ~80ns and lots of tracks end up getting rejected.

Maybe @brownd1978 can confirm if I'm doing this correctly? Maybe `TwoLinePCA` should also have an `LRambig()` function.

Reply:
Right, I remember now - I wasn't sure if this was correct, I thought s1 was the distance along the straw direction, i.e. whether the track was on the Cal or HV side of the straw

Reply:
Ah... this might need some checking then. In this snippet 
https://github.com/Mu2e/Offline/blob/bd2e7884502edfa02712645a29f482fb11431a2f/CosmicReco/src/DriftFitUtils.cc#L119-L138

The track information is passed before the straw information. s1() and s2() are different if you swap the order around like this
https://github.com/Mu2e/Offline/blob/bd2e7884502edfa02712645a29f482fb11431a2f/CosmicReco/src/PDFFit.cc#L302-L304
...so I'm not sure which is right. Maybe the best thing to do is just to be consistent?

Reply:
I dont think s2 is right either, since thats just which side of the track startposition you are on - I think you would need to do something like https://github.com/Mu2e/Offline/blob/master/TrkDiag/src/KalDiag.cc#L481

Reply:
I followed Dave's suggestion from earlier.  Happy to try this method as well

Reply:
I wrote a HitAmbiguity function implementing the suggested method:
https://github.com/Mu2e/Offline/blob/d0db6280233b105ec5e8f2b9f746926253edf01f/CosmicReco/src/PDFFit.cc#L375-L393

Does this make sense? I've not tested it yet.